scripts/DeleteIndicesPattern.js

```diff
@@ -0,0 +1,117 @@
+const algolia = require('algoliasearch');
+const HttpsAgent = require('agentkeepalive').HttpsAgent;
+const keepaliveAgent = new HttpsAgent({
+  maxSockets: 1,
+  maxKeepAliveRequests: 0, // no limit on max requests per keepalive socket
+  maxKeepAliveTime: 30000, // keepalive for 30 seconds
+});
+const readLine = require('readline');
+const Base = require('./Base.js');
+
+class DeleteIndicesPatternScript extends Base {
+  constructor() {
+    super();
+    // Define validation constants
+    this.message =
+      "\nUsage: $ algolia deleteindices -a algoliaappid -k algoliaapikey -r 'regexp for filtering' -x\n\n";
+    this.params = ['algoliaappid', 'algoliaapikey', 'regexp', 'dryrun'];
```

Comment:
I know it says in readme it is required, but doesn't it default to `false`?

Reply:
Ideally this is what I wanted to do:
```sh
algolia deleteindicespattern -a -k -r
```

^ This would by default be in dry run

Then at the end of the output, we would put:

```sh
# Once you have reviewed the dry run, you can run it for real with:
algolia deleteinidicespattern -a -k -r --no-dryrun
```

But I never found the exact way to achieve that with commander.js and the current project architecture, can you help?

Reply:
Just unresolved to make sure we address that if that's ok, not sure how to do it really

Reply:
Cool 👍

Reply:
I think to allow flag that take no options, it seems like we'd do almost the same thing we are for flags _with_ options... [see here](https://github.com/tj/commander.js/#option-parsing)

I think it could be OK for us to merge this feature as-is, as I think that on subsequent PRs immediately following this one I'll be looking to refactor how we're using commander.js... specifically I'd like to make the `index.js` entry point more modular, eg. allowing sub-commands, allowing unique flags/options for individual commands (unlike today how all options are shared between commands), etc.

What do you think? Shall we merge this (also, will we pin event-stream version to 3.3.4?) and then I start work on the refactor?

Reply:
I am fine merging it as if, it works, but before releasing it we might want to allow for the variant in https://github.com/algolia/algolia-cli/pull/16#discussion_r240205177 if you find out how to do it yep! Unsure how much of a refactor it is.

As for event-stream 4.x is really not impacted 100% sure here it's safe